runtime/vm/lookupmethod.c

```diff
@@ -1031,10 +1031,84 @@ illegalAccessMessage(J9VMThread *currentThread, IDATA badMemberModifier, J9Class
 	char *destModuleMsg = NULL;
 	char packageNameBuf[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
 	char *packageNameMsg = NULL;
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	J9Class *unverifiedNestMemberClass = NULL;
+#endif /* J9VM_OPT_VALHALLA_NESTMATES */
 	
 	PORT_ACCESS_FROM_VMC(currentThread);
 	Trc_VM_illegalAccessMessage_Entry(currentThread, J9UTF8_LENGTH(senderClassNameUTF), J9UTF8_DATA(senderClassNameUTF),
 			J9UTF8_LENGTH(targetClassNameUTF), J9UTF8_DATA(targetClassNameUTF), badMemberModifier);
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	/* Values of error codes for nestmates visibility checks on source class is
+	 * odd and on destination/target class is even
+	 */
+	if ((J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_SOURCE_CLASS <= errorType)
+			&& (J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR_DEST_CLASS >= errorType)) {
+		if (1 == (errorType & 1)) {
+			unverifiedNestMemberClass = targetClass;
+		} else  {
+			unverifiedNestMemberClass = senderClass;
+		}
+	}
+
+	/* If an issue with the nest host loading and verification occurred, then it will be one of:
+	 * 		J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_SOURCE_CLASS
+	 * 		J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_DEST_CLASS
+	 * 		J9_VISIBILITY_NEST_HOST_DIFFERENT_PACKAGE_ERROR_SOURCE_CLASS
+	 * 		J9_VISIBILITY_NEST_HOST_DIFFERENT_PACKAGE_ERROR_DEST_CLASS
+	 * 		J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR_SOURCE_CLASS
+	 * 		J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR_DEST_CLASS
+	 * Otherwise, it is one of:
+	 * 		J9_VISIBILITY_NON_MODULE_ACCESS_ERROR
+	 * 		J9_VISIBILITY_MODULE_READ_ACCESS_ERROR
+	 * 		J9_VISIBILITY_MODULE_PACKAGE_EXPORT_ERROR
+	 */
+	if (NULL != unverifiedNestMemberClass) {
+		J9ROMClass *romClass = unverifiedNestMemberClass->romClass;
+		J9UTF8 *nestMemberNameUTF = J9ROMCLASS_CLASSNAME(romClass);
+		J9UTF8 *nestHostNameUTF = J9ROMCLASS_NESTHOSTNAME(romClass);
+
+		if ((J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_SOURCE_CLASS == errorType)
```

Comment:
This is another opportunity to use a switch.  It may or may not be cleaner here.